MaterialDesignThemes.Wpf/Extensions.cs

```diff
@@ -41,6 +41,28 @@ public static IEnumerable<DependencyObject> GetVisualAncestory(this DependencyOb
                 yield return leaf;
                 leaf = VisualTreeHelper.GetParent(leaf);
             }
-        } 
+        }
+
+        public static IEnumerable<DependencyObject> GetLogicalAncestory(this DependencyObject leaf)
+        {
+            while (leaf != null)
+            {
+                yield return leaf;
+                leaf = LogicalTreeHelper.GetParent(leaf);
+            }
+        }
+
+        public static bool HasAncestor(this DependencyObject leaf, DependencyObject ancestor)
+        {
+            var visualAncestry = leaf.GetVisualAncestory().ToArray();
```

Comment:
How about this, otherwise we are always going to iterating visual ancestry at least once; for the array, and then possible the whole array again.

```
            DependencyObject parent = null;
            foreach (var node in leaf.GetVisualAncestory())
            {
                if (Equals(node, ancestor))
                    return true;

                parent = node;
            }

            return parent.GetLogicalAncestory().Contains(ancestor);
```

(havent tested that code)

and damn...just realist I am spelling ancestry wrong in the code :worried: